integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java

```diff
@@ -567,28 +573,50 @@ private int writeInternal(String path, ByteBuffer buf, long size, long offset, F
         }
       }
     }
-
-    FileOutStream os = ce.getOut();
-    long bytesWritten = os.getBytesWritten();
-    if (offset != bytesWritten && offset + sz > bytesWritten) {
-      LOG.error("Only sequential write is supported. Cannot write bytes of size {} to offset {} "
-          + "when {} bytes have written to path {}", size, offset, bytesWritten, path);
-      return -ErrorCodes.EIO();
-    }
-    if (offset + sz <= bytesWritten) {
-      LOG.warn("Skip writting to file {} offset={} size={} when {} bytes has written to file",
-          path, offset, sz, bytesWritten);
-      // To fulfill vim :wq
-      return sz;
-    }
-
-    try {
-      final byte[] dest = new byte[sz];
-      buf.get(dest, 0, sz);
-      os.write(dest);
-    } catch (IOException e) {
-      LOG.error("IOException while writing to {}.", path, e);
-      return -ErrorCodes.EIO();
+    if (!mTmpFolder.isEmpty()) {
+      RandomAccessFile tmpFile;
+      try {
+        File tmpFolder = new File(mTmpFolder);
+        if (!tmpFolder.exists()) {
+          tmpFolder.mkdirs();
```

Comment:
why we put the creation of this temp dir here inside the critical path of `writeInternal` method?
this is only needed once and in this way each `writeInternal` call (could be many) will check existence and make it slow.
can we move it to constructor to prepare the temp dir only once if this is enabled?